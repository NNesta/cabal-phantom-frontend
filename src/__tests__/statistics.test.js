import Content from '../components/dashboard/Homepage/Content';
import React from 'react';
import { render, fireEvent, cleanup, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import userEvent from '@testing-library/user-event';
import { BrowserRouter as Router, Link } from 'react-router-dom';


const ContentComponent = () => {
	return render(
		<Provider store={store}>
				<Router>
					<Content />
				</Router>
		</Provider>
	);
};

describe('TopBar component', () => {
    afterEach(cleanup)

    it('Should render add user component properly', () => {
		const { asFragment } = ContentComponent();
		expect(asFragment(<Content />)).toMatchSnapshot();
	});
})
