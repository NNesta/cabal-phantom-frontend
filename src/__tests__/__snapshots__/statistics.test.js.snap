// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar component Should render add user component properly 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
  >
    <div
      class="d-flex flex-wrap row"
    >
      <div
        class="mt-4 col-sm-6 col-lg-3"
      >
        <div
          class="card card-body"
        >
          <div
            class="d-flex align-items-start"
          >
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="font-size: 3rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
                />
              </svg>
            </div>
            <div
              class="ml-4"
            >
              <h2
                class="card-title"
              >
                Visitors
              </h2>
              <p
                class="card-text"
                style="font-size: 1.5rem; color: rgb(220, 53, 69);"
              >
                <span>
                  1000
                </span>
                <span
                  class="ml-5"
                >
                  People
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-4 col-sm-6 col-lg-3"
      >
        <div
          class="card card-body"
        >
          <div
            class="d-flex align-items-start"
          >
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="font-size: 3rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17 5H3c-1.1 0-2 .89-2 2v9h2c0 1.65 1.34 3 3 3s3-1.35 3-3h5.5c0 1.65 1.34 3 3 3s3-1.35 3-3H23v-5l-6-6zM3 11V7h4v4H3zm3 6.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm7-6.5H9V7h4v4zm4.5 6.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM15 11V7h1l4 4h-5z"
                />
              </svg>
            </div>
            <div
              class="ml-4"
            >
              <h2
                class="card-title"
              >
                Buses
              </h2>
              <p
                class="card-text"
                style="font-size: 1.5rem; color: rgb(220, 53, 69);"
              >
                <span>
                  200
                </span>
                <span
                  class="ml-5"
                >
                  Buses
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-4 col-sm-6 col-lg-3"
      >
        <div
          class="card card-body"
        >
          <div
            class="d-flex align-items-start"
          >
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="font-size: 3rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2c-4.42 0-8 .5-8 4v9.5C4 17.43 5.57 19 7.5 19L6 20.5v.5h12v-.5L16.5 19c1.93 0 3.5-1.57 3.5-3.5V6c0-3.5-3.58-4-8-4zM7.5 17c-.83 0-1.5-.67-1.5-1.5S6.67 14 7.5 14s1.5.67 1.5 1.5S8.33 17 7.5 17zm3.5-6H6V6h5v5zm5.5 6c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm1.5-6h-5V6h5v5z"
                />
              </svg>
            </div>
            <div
              class="ml-4"
            >
              <h2
                class="card-title"
              >
                Routes
              </h2>
              <p
                class="card-text"
                style="font-size: 1.5rem; color: rgb(220, 53, 69);"
              >
                <span>
                  50
                </span>
                <span
                  class="ml-5"
                >
                  Routes
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mt-4 col-sm-6 col-lg-3"
      >
        <div
          class="card card-body"
        >
          <div
            class="d-flex align-items-start"
          >
            <div
              class="icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                style="font-size: 3rem;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
                />
              </svg>
            </div>
            <div
              class="ml-4"
            >
              <h2
                class="card-title"
              >
                Users
              </h2>
              <p
                class="card-text"
                style="font-size: 1.5rem; color: rgb(220, 53, 69);"
              >
                <span>
                  550
                </span>
                <span
                  class="ml-5"
                >
                  People
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="mt-4 col-sm-12"
      >
        <div
          class="p-5 card card-body"
        >
          <h1
            class="card-title"
          >
            Statistics
          </h1>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="mt-4 col-sm-12 col-lg-6"
      >
        <div
          class="p-5 card card-body"
        >
          <h1
            class="card-title"
          >
            Most Travelled Routes
          </h1>
        </div>
      </div>
      <div
        class="mt-4 col-sm-12 col-lg-6"
      >
        <div
          class="p-5 card card-body"
        >
          <h1
            class="card-title"
          >
            Most Active Buses
          </h1>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
